lib/plausible_web/live/goal_settings/form.ex

```diff
@@ -33,13 +43,15 @@ defmodule PlausibleWeb.Live.GoalSettings.Form do
         event_name_options: Enum.map(assigns.event_name_options, &{&1, &1}),
         current_user: assigns.current_user,
         domain: assigns.domain,
-        selected_tab: "custom_events",
+        selected_tab: selected_tab,
         tab_sequence_id: 0,
         site: site,
         has_access_to_revenue_goals?: has_access_to_revenue_goals?,
         existing_goals: assigns.existing_goals,
         on_save_goal: assigns.on_save_goal,
-        on_autoconfigure: assigns.on_autoconfigure
+        on_autoconfigure: assigns.on_autoconfigure,
+        goal_id: goal_id,
+        goal: goal_id && goal
```

Comment:
```suggestion
        goal: goal
```

should be good enough